admin/modules/forum/announcements.php

```diff
@@ -589,23 +589,23 @@ function checkAction(id)
 		}
 
 		// Gather start and end date data
-		$startday = $mybb->input['starttime_day'];
+		$startday = $mybb->get_input('starttime_day', MyBB::INPUT_INT);
 		$start_time = $mybb->input['starttime_time'];
-		$startmonth = $mybb->input['starttime_month'];
+		$startmonth = $mybb->get_input('starttime_month', MyBB::INPUT_INT);
 		$startmonthsel[$startmonth] = 'selected="selected"';
-		$startdateyear = $mybb->input['starttime_year'];
+		$startdateyear = $mybb->get_input('starttime_year', MyBB::INPUT_INT);
 
 		if($mybb->input['endtime_type'] == 1)
 		{
 			// Set time
 			$endtime_checked[1] = 'checked="checked"';
 			$endtime_checked[2] = '';
 
-			$endday = $mybb->input['endtime_day'];
+			$endday = $endmonth = $mybb->get_input('endtime_day', MyBB::INPUT_INT);
```

Comment:
```suggestion
			$endday = $mybb->get_input('endtime_day', MyBB::INPUT_INT);
```

If there is any undefined variable warning then we can fix that later.